main.cpp

```diff
@@ -51,23 +51,438 @@ If you need inspiration for what to write, take a look at previously approved st
+furColor("black"),
+entranceFee(1500.0f),
+totalWinnings(4500.0f)
+{
+    std::cout << "Dog constructor called" << std::endl;
+}
+
+DogShow::~DogShow()
+{
+    std::cout << "Dog destructor called" << std::endl;
+}
+
+float DogShow::calculateCostToEnterDogShow(int weightInPounds, std::string furColor, float entranceFee)
+{
+    int totalCost = entranceFee;
+    if(weightInPounds > 50)
+    {
+        totalCost += 50.45;
+    }
+    else if(furColor == "white")
+    {
+        totalCost+= 75.63;
+    }
+
+    else
+    {
+        std::cout << "No additional costs" << std::endl;
+    }
+
+    return totalCost;
+
+}
+
+
+void DogShow::predictWinProbabilityDogShow(std::string breedName, int weightInPounds, std::string furColor)
+{
+    float winProbability  = 0.0f;
+
+    std::string breeds[5] = {"Pitbull", "German Shepherd", "Lab", "Great Dane", "Mastiff"};
+
+    for(int i = 0; i < 5; i++)
+    {
+        if(breedName == breeds[i])
+        {
+            winProbability += .4;
+            break;
+        }
+    }
+
+    if(furColor != "white")
+    {
+        winProbability += .2;
+    }
+
+    winProbability = (winProbability+ weightInPounds) / 100;
+
+    std::cout << winProbability << "chance of win probability" << std::endl;
+
+}
+
+float DogShow::calculateTotalEarningsDogShow(float totalWinnings, float entranceFee)
+{
+    return (totalWinnings - entranceFee);
+}
+
+DogShow::Animal::Animal():
+theoreticalSpeed(45),
+animalType("dog"),
+countryOfShow("USA"),
+ageOfAnimal(10)
+{
+    std::cout << "Animal constructor called" << std::endl;
+}
+
+DogShow::Animal::~Animal()
+{
+    std::cout << "Animal Destructor called" << std::endl;
+}
+
+bool DogShow::Animal::determineAnimalShow(std::string animalType, std::string countryOfShow)
+{
+    if((animalType == "dog" || animalType == "horse") && (countryOfShow == "USA"))
+    {   
+        std::cout << "Show exsists for" << animalType << std::endl;
+        return true;
+    }
+
+    std::cout << "No show exists for animal" << animalType << std::endl;
+    return false;
+}
+
+int DogShow::Animal::calculateMaxLandSpeed(int theoreticalSpeed)
+{
+    return (theoreticalSpeed - 10);
+}
+
+bool DogShow::Animal::determineAnimalHealth(std::string animalType, int ageOfAnimal)
+{
+    if(ageOfAnimal > 10)
+    {
+        std::cout << animalType << "health is bad" << std::endl;
+        return false;
+    }
+
+    std::cout << animalType << "health is good" << std::endl;
+    return true;
+}
+
 
 /*
  new UDT 4:
  */
+struct GermanShepherd
+{
+    struct DogShow danielsDog;
+
+    void calculateGermanShepherdEarnings();
+    void finishObstacleCourse();
+
+    GermanShepherd();
+    ~GermanShepherd();
+
+};
+
+GermanShepherd::GermanShepherd()
+{
+    std::cout << "German Shepherd constructor" << std::endl;
+}
+GermanShepherd::~GermanShepherd()
+{
+    std::cout << "German Sheperd destructor" << std::endl;
+}
+
+void calculateGermanShepherdEarnings()
+{
+
+}
+
+void finishObstacleCourse()
+{
+
+}
```

Comment:
These are free functions, not member functions
Implement something in them.
- [ ] revise